JOG 107 sessions in dem history tab anzeigen

JGlumanda opened the pull request:
Added method to session_manager.dart
--> getRunTimeAtTimestamp()
--> getSpeeds()
--> Edited Method getTopSpeed() to round value 2 digits after comma and returns the string
--> Edited getEndTimeAsString() and getStartTimeAsString() format to 'HH:mm' because no one cares about seconds

Edited main.dart and all other main pages
--> Moved Padding from every Page to now only appear in main.dart

Edited page_history.dart
--> It now shows the saved sessions
--> Shows date and time room run on the Card

Edited page_run_insights.dart
--> Shows all info to the session
--> Shows a chart of the speeds that the user run

Removed run.dart because it is not needed anymore

Added package syncfusion_flutter_charts

![Screenshot_1677000491](https://user-images.githubusercontent.com/94634381/220417274-f7524671-a22b-48bf-b8a0-1f12c691a1e3.png)
![Screenshot_1677000494](https://user-images.githubusercontent.com/94634381/220417276-bd1b26c8-5c6d-40dc-960f-d2e8907f25b5.png)
![Screenshot_1677000496](https://user-images.githubusercontent.com/94634381/220417280-199d6a30-1b3f-44e3-ab92-3706e2ca78d8.png)

JGlumanda committed:
Added method to session_manager.dart
--> getRunTimeAtTimestamp()
--> getSpeeds()
--> Edited Method getTopSpeed() to round value 2 digits after comma and returns the string
--> Edited getEndTimeAsString() and getStartTimeAsString() format to 'HH:mm' because no one cares about seconds

Edited main.dart and all other main pages
--> Moved Padding from every Page to now only appear in main.dart

Edited page_history.dart
--> It now shows the saved sessions
--> Shows date and time room run on the Card

Edited page_run_insights.dart
--> Shows all info to the session
--> Shows a chart of the speeds that the user run

Removed run.dart because it is not needed anymore

Added package syncfusion_flutter_charts

Signed-off-by: JGlumanda <jonas-berger@hotmail.com>

JGlumanda committed:
The sessions are now initially getting loaded when you open the app

Signed-off-by: JGlumanda <jonas-berger@hotmail.com>

nikho2000 commented on lib/main.dart:
```diff
@@ -6,13 +6,15 @@ import 'package:jog_dog/pages/page_home.dart';
 void main() {
   runApp(const MyApp());
   requestPermissions();
+  SessionManager().loadSessionsFromJson();
```
Gibt es einen Grund weswegen du hier den SessionManager initiallisierst und dann direkt im anschluss nochmal loadSessionsFromJson() aufrufst? Das wird initial wenn der Singelton vom SessionManager erstellt wird automatisch erledigt. Effektiv wird die loadSessionsFromJson Methode hier 2 mal ausgeführt, direkt hintereinander.

JGlumanda replied:
Es ist keine Initialisierung da ich hier nur auf die Instanz zugreife.
Dazu mache ich dass da ich so garantieren kann dass die Sessions geladen sind wenn ich das erstmal mal im History Tab bin, das konnte ich davor nicht. Davor ist es so passiert dass ich in den History Tab gegangen bin und die Sessions nicht geladen waren obwohl welche gespeichert waren

JGlumanda committed:
The latest session is now getting displayed first in the history page

Start Run button now properly displays if a run is started or not

You can stop a session

Sessions can be deleted when holding long on the Card

Signed-off-by: JGlumanda <jonas-berger@hotmail.com>

JGlumanda committed:
Added a message when no session is found

Signed-off-by: JGlumanda <jonas-berger@hotmail.com>

JGlumanda committed:
Session now getting deleted from session list if deleted

Signed-off-by: JGlumanda <jonas-berger@hotmail.com>

nikho2000 commented on lib/utilities/session_manager.dart:
```diff
@@ -111,14 +115,16 @@ class SessionManager {
+  /// Deletes the session with the given id from the local storage and the list
   void deleteSession(String id) {
     SessionFileManager().deleteSession(id);
+    sessions.removeWhere((session) => session.id == id);
```
sessions zu _sessions bitte abändern.

GoGoWe commented on lib/pages/page_history.dart:
```diff
@@ -11,52 +10,90 @@ class History extends StatefulWidget {
+                        onLongPress: () {
+                          setState(
+                            () {
+                              showDialog<void>(
```
Ich würde zumindest das `showDialog ` in ein extra Widget auslagern weiter unten in der Datei, sonst ist das echt etwas unleserlich (Ist auch die Empfehlung von Google) 
Dann steht auch das onTab da nicht so im nirgendwo

GoGoWe commented on lib/pages/page_home.dart:
```diff
@@ -21,23 +21,20 @@ class _HomeState extends State<Home> {
+            const SizedBox(
+              height: 15,
+            ),
+            const SpotifyButton(),
```
der Button sollte wirklich mal umbenannt werden z.B. connectMusicButton()

JGlumanda committed:
Edited _isRunning in startsessionbutton.dart

Edited sessions.remove to _sessions.remove in session_manager.dart

Signed-off-by: JGlumanda <jonas-berger@hotmail.com>

JGlumanda committed:
Ups

Signed-off-by: JGlumanda <jonas-berger@hotmail.com>

JGlumanda committed:
Changed something with permissions

Signed-off-by: JGlumanda <jonas-berger@hotmail.com>

JGlumanda committed:
Merge remote-tracking branch 'origin/develop' into JOG-107-sessions-in-dem-history-tab-anzeigen

# Conflicts:
#	lib/main.dart
#	lib/pages/page_history.dart
#	lib/pages/page_run_insights.dart
#	lib/pages/page_settings.dart
#	lib/widgets/startsessionbutton.dart

JGlumanda committed:
removed some conflicts that occurred because of the merge

Signed-off-by: JGlumanda <jonas-berger@hotmail.com>

JGlumanda committed:
Fixed Bug that you can read from file if it does not exist

Signed-off-by: JGlumanda <jonas-berger@hotmail.com>

JGlumanda committed:
Fixed Bug that Files are trying to get listed even if the path does not exist

reverted change that padding from a page is set in main.dart

Signed-off-by: JGlumanda <jonas-berger@hotmail.com>

JGlumanda committed:
outsourced delete Session Popup

Signed-off-by: JGlumanda <jonas-berger@hotmail.com>

GoGoWe commented on lib/pages/page_history.dart:
```diff
@@ -10,58 +10,91 @@ class History extends StatefulWidget {
+                          onLongPress: () {
+                            setState(
+                              () {
+                                showDialog<void>(
```
Still not extracted?

GoGoWe reviewed (approved):
Alles super es muss noch etwas refactored werden kann aber im nachhinein gemacht werden
